std/algorithm/iteration.d

```diff
@@ -1524,14 +1540,26 @@ private struct GroupByImpl(alias pred, Range)
     }
 
     @property bool empty() { return impl.current.empty; }
-    @property auto front() { return Group(impl); }
+
+    @property auto front()
+    {
+        static if (isUnary)
+        {
+            import std.typecons : tuple;
+            return tuple(unaryFun!pred(impl.current.front), Group(impl));
```

Comment:
But sure, the return type is ready for release. Internal optimization could be post-phoned. Still, I'd rather see it perfect. ;)